File: android/src/main/kotlin/com/tsitser/background_http_plugin/domain/usecase/CreateRequestUseCase.kt
package com.tsitser.background_http_plugin.domain.usecase

import com.tsitser.background_http_plugin.domain.entity.HttpRequest
import com.tsitser.background_http_plugin.domain.entity.TaskInfo
import com.tsitser.background_http_plugin.domain.repository.TaskRepository

/**
 * Use case for creating HTTP request
 */
class CreateRequestUseCase(
    private val repository: TaskRepository
) {
    suspend operator fun invoke(request: HttpRequest): TaskInfo {
        return repository.createTask(request)
    }
}

File: ios/Classes/Domain/UseCase/CreateRequestUseCase.swift
import Foundation

/// Use case for creating HTTP request
struct CreateRequestUseCase {
    private let repository: TaskRepository
    
    init(repository: TaskRepository) {
        self.repository = repository
    }
    
    func execute(request: HttpRequest) async throws -> TaskInfo {
        return try await repository.createTask(request: request)
    }
}

